test/read.jl

```diff
@@ -0,0 +1,245 @@
+mktempdir() do dir
+
+tasks = []
+
+# Create test file...
+filename = joinpath(dir, "file.txt")
+text = "C1,C2\n1,2\na,b\n"
+
+# List of IO producers...
+l = Vector{Tuple{AbstractString,Function}}()
+
+
+# File
+io = (text) -> begin
+    open(io-> write(io, text), filename, "w")
+    Base.Filesystem.open(filename, Base.Filesystem.JL_O_RDONLY)
+end
+s = io(text)
+@test isa(s, IO)
+@test isa(s, Base.Filesystem.File)
+close(s)
+push!(l, ("File", io))
+
+
+# IOStream
+io = (text) -> begin
+    open(io-> write(io, text), filename, "w")
+    open(filename)
+end
+s = io(text)
+@test isa(s, IO)
+@test isa(s, IOStream)
+close(s)
+push!(l, ("IOStream", io))
+
+
+# IOBuffer
+io = (text)->IOBuffer(text)
+s = io(text)
+@test isa(s, IO)
+@test isa(s, IOBuffer)
+close(s)
+push!(l, ("IOBuffer", io))
+
+
+function run_test_server(srv, text)
+    push!(tasks, @async begin
+        try
+            sock = accept(srv)
+            try
+                write(sock,text)
+            catch e
+                if typeof(e) != Base.UVError
+                    rethrow(e)
+                end
+            finally
+                close(sock)
+            end
+        finally
+            close(srv)
+        end
+    end)
+end
+
+if !haskey(ENV, "TRAVIS")
```

Comment:
ahk. did you track it down? I'm finding TCPSockets kind of challenging to add tests for myself. This branch helps get MbedTLS working again, though I'm not sure what its impacts are on the optimizations Jameson had been working towards.